aptos-move/aptos-release-builder/src/components/mod.rs

```diff
@@ -44,16 +58,21 @@ pub enum ReleaseEntry {
     Version(Version),
     FeatureFlag(Features),
     Consensus(OnChainConsensusConfig),
+    RawScript(PathBuf),
 }
 
 impl ReleaseEntry {
     pub fn generate_release_script(
         &self,
         client: Option<&Client>,
         result: &mut Vec<(String, String)>,
-        is_testnet: bool,
-        is_multi_step: bool,
+        execution_mode: ExecutionMode,
     ) -> Result<()> {
+        let (is_testnet, is_multi_step) = match execution_mode {
+            ExecutionMode::MultiStep => (false, true),
```

Comment:
Can we deprecate or make automatic multi vs single step?

Can the tool just know whether to execute multi or single

Reply:
I would like to keep them manual for now just to see how we are planning to use this feature in the longer run. It seems that we always need multi-step for framework upgrade but for others, if there's only one entry we can use single step.